docs/operate-and-deploy/installation/upgrading.md

```diff
@@ -42,6 +42,30 @@ upgrading a cluster involves leaving the old cluster running on the old version,
 cluster on the new version, porting across your database schema, and finally thinking about your data.
 Read on for details.
 
+
+## (1) In-Place upgrade
+
+#### Steps :-
+
+1. Download the [new ksqlDB version](https://docs.confluent.io/platform/current/installation/installing_cp/zip-tar.html#get-the-software) to upgrade to.
+2. Set it up with all required configs. In our case we mostly copied all configs from the current version(which is running) to the new version.
+3. Take Backup and then copy all UDF JAR to new version .(OPTIONAL STEP In case you don't have any UDF).
```

Comment:
```suggestion
3. Copy source UDFs and recompile against the new ksqlDB version, to accommodate any changes to the UDF framework.
```
Recompiling UDFs is not necessary for most upgrades, but sometimes it is so it doesn't hurt to include a mention in these upgrade steps.